common/src/app/common/logic/libraries.cljc

```diff
@@ -1624,6 +1624,56 @@
       (seq applied-tokens)
       (update-tokens container dest-shape origin-shape applied-tokens))))
 
+(defn- generate-update-attrs
+  [changes dest-shape container roperations uoperations]
+  (let [all-parents (cfh/get-parent-ids (:objects container)
+                                        (:id dest-shape))]
+    (-> changes
+        (update :redo-changes conj (make-change
+                                    container
+                                    {:type :mod-obj
+                                     :id (:id dest-shape)
+                                     :operations roperations}))
+        (update :redo-changes conj (make-change
+                                    container
+                                    {:type :reg-objects
+                                     :shapes all-parents}))
+        (update :undo-changes conj (make-change
+                                    container
+                                    {:type :mod-obj
+                                     :id (:id dest-shape)
+                                     :operations (vec uoperations)}))
+        (update :undo-changes concat [(make-change
+                                       container
+                                       {:type :reg-objects
+                                        :shapes all-parents})]))))
+
+(defn- update-attr-operations
```

Comment:
Here I suggest `add-update-attr-operations`, to start with a verb and be consistent with previous one.